### PR TITLE
Add several fresh papers

### PR DESCRIPTION
**Pull request by EgorBu**
* Learning Compositional Neural Programs with Recursive Tree Search and Planning
* Program Synthesis and Semantic Parsing with Learned Code Idioms
* Modeling Vocabulary for Big Code Machine Learning
* Towards Neural Decompilation

Signed-off-by: egor <egor@sourced.tech>

Please fill the `[ ]` with an `x` to make sure that you read and followed the checklist

- [x] This addition is about Machine Learning __on Code__ specifically. General links should be added to [Awesome Machine Learning](https://github.com/josephmisiti/awesome-machine-learning) instead.
- [x] I have checked that the proposed addition is not already in the awesome list.
- [x] I have signed-off my commits as detailed [here](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md).
- [x] The links follow the link template:

        * [Title](URL) - Description.
    or

        * [Title](URL) - Description [Code](link to code).
- [x] The addition is correctly classified as a paper, blog post, talk, software or dataset.
- [x] The commit message must start with a verb (`Add …`, `Deprecate …`) and end without a dot. For exemple:

        Deprecate Theano software

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,6 +72,7 @@ A curated list of awesome research papers, datasets and software projects devote
 
 #### Program Synthesis and Induction
 
+- <img src="badges/12-pages-gray.svg" alt="16-pages" align="top"> [Program Synthesis and Semantic Parsing with Learned Code Idioms](https://arxiv.org/abs/1906.10816v2) - Richard Shin, Miltiadis Allamanis, Marc Brockschmidt, Oleksandr Polozov, 2019.
```
Comment (m09) [inline README.md]:
```suggestion
- <img src="badges/12-pages-gray.svg" alt="12-pages" align="top"> [Program Synthesis and Semantic Parsing with Learned Code Idioms](https://arxiv.org/abs/1906.10816v2) - Richard Shin, Miltiadis Allamanis, Marc Brockschmidt, Oleksandr Polozov, 2019.
```

File: README.md [hunk 2]
```diff
@@ -107,6 +108,7 @@ A curated list of awesome research papers, datasets and software projects devote
 
 #### Source Code Analysis and Language modeling
 
+- <img src="badges/12-pages-gray.svg" alt="24-pages" align="top"> [Modeling Vocabulary for Big Code Machine Learning](https://arxiv.org/abs/1904.01873v1) - Hlib Babii, Andrea Janes, Romain Robbes, 2019.
```
Comment (m09) [inline README.md]:
```suggestion
- <img src="badges/12-pages-gray.svg" alt="12-pages" align="top"> [Modeling Vocabulary for Big Code Machine Learning](https://arxiv.org/abs/1904.01873v1) - Hlib Babii, Andrea Janes, Romain Robbes, 2019.
```

File: README.md [hunk 3]
```diff
@@ -125,6 +127,7 @@ A curated list of awesome research papers, datasets and software projects devote
 
 #### Neural Network Architectures and Algorithms
 
+- <img src="badges/19-pages-gray.svg" alt="11-pages" align="top"> [Learning Compositional Neural Programs with Recursive Tree Search and Planning](https://arxiv.org/abs/1905.12941v1) - Thomas Pierrot, Guillaume Ligner, Scott Reed, Olivier Sigaud, Nicolas Perrin, Alexandre Laterre, David Kas, Karim Beguir, Nando de Freitas, 2019.
```
Comment (m09) [inline README.md]:
```suggestion
- <img src="badges/19-pages-gray.svg" alt="19-pages" align="top"> [Learning Compositional Neural Programs with Recursive Tree Search and Planning](https://arxiv.org/abs/1905.12941v1) - Thomas Pierrot, Guillaume Ligner, Scott Reed, Olivier Sigaud, Nicolas Perrin, Alexandre Laterre, David Kas, Karim Beguir, Nando de Freitas, 2019.
```

File: README.md [hunk 4]
```diff
@@ -160,6 +163,7 @@ A curated list of awesome research papers, datasets and software projects devote
 
 #### Program Translation
 
+- <img src="badges/18-pages-gray.svg" alt="14-pages" align="top"> [Towards Neural Decompilation](https://arxiv.org/abs/1905.08325v1) - Omer Katz, Yuval Olshaker, Yoav Goldberg, Eran Yahav, 2019.
```
Comment (m09) [inline README.md]:
```suggestion
- <img src="badges/18-pages-gray.svg" alt="18-pages" align="top"> [Towards Neural Decompilation](https://arxiv.org/abs/1905.08325v1) - Omer Katz, Yuval Olshaker, Yoav Goldberg, Eran Yahav, 2019.
```

**Review by m09 — CHANGES_REQUESTED** [review]
Nice papers :+1:

File: README.md (hunk 2 above)
Comment (EgorBu) [reply README.md]: completely forgot about a second place with a number of pages - nicely spotted)

**Review by m09 — APPROVED** [review]
:rocket:

**Pull request merged**